runtime/lua/vim/lsp/buf.lua

```diff
@@ -192,17 +192,25 @@ end
 ---     - range (table|nil) Range to format.
 ---         Table must contain `start` and `end` keys with {row,col} tuples using
 ---         (1,0) indexing.
+---         Can also be a list of tables that contain `start` and `end` keys as described above,
```

Comment:
It'd be better to be specific. Although the previous doc (written #19998) is also ambiguous, but it'd be better to fix in this opportunity. For normal range formatting request, we can mention `lsp.Range`; for the ranges formatting request, it would be `lsp.Range[]`.

Reply:
Or we could name the parameter as `ranges` instead of `range` if we'd want to make it closer to the type defined in the LSP protocol (see `DocumentRangesFormattingParams`), with more strict type checking, but I don't have a string opinion here in the API design. Any thoughts, @MariaSolOs @mfussenegger?

Reply:
I agree that using `lsp.Range` here is nicer/safer. I don't have any preference regarding naming.

Reply:
Hmm the thing is that this is **not** an LSP range

The function expects a range of the form  `{ start = {row, col}, end = {row, col} }` with `row` being 1-indexed, but `col` being 0-indexed (for some reason...?)

Meanwhile, an [LSP range](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#range) is `{ start = { line = .., character = ..}, end = { line =..., character = ...}}`, with both `line` and `character` using 0-indexing.

Only the call to `utils.make_given_range_params` turns our "custom" range into a LSP range.

I'd also find it more intuitive if this would take a LSP range instead, but I think there are two problems:

1) API break (obviously)
2) we don't have a (convenient) API yet to create LSP range objects, I think...? The closest match is `make_given_range_params`, which returns (as the name says) a `RangeParams` LSP object, not a `Range`. If we're fine with the API break, I think we should add a `vim.lsp.util.make_give_range`

Reply:
I see. My bad, sorry for confusing. (This is the reason why it has to be more clearly documented!) You're right -- the parameter isn't already a LSP range, but *(1,0)-indexed* API (row, col) representation; then we should not change and break the API.

Reply:
Yeah, https://github.com/neovim/neovim/issues/25509 is a tracking issue for this problem.

Reply:
Right, I'll add a comment along the lines of (this is **not** an LSP range) to the docs, if everyone's fine with that

Reply:
I don't think such a remark is needed. The longer the docs are, the less chances people will actually read them. I think that the current description is good enough.